classes/issue/IssueDAO.inc.php

```diff
@@ -586,12 +586,12 @@ function getByPubIdType($pubIdType, $contextId = null, $pubIdSettingName = null,
 				LEFT JOIN issue_settings ist ON (i.issue_id = ist.issue_id)
 				'. ($pubIdSettingName != null?' LEFT JOIN issue_settings iss ON (i.issue_id = iss.issue_id AND iss.setting_name = ?)':'') .'
 			WHERE
-				ist.setting_name = ? AND ist.setting_value IS NOT NULL
+				i.published = 1 AND ist.setting_name = ? AND ist.setting_value IS NOT NULL
```

Comment:
yes, I don't understand it either, but it is so. For example this query, searching for unregistered issues:
```
SELECT i.*
            FROM issues i
                LEFT JOIN custom_issue_orders o ON (o.issue_id = i.issue_id)
                LEFT JOIN issue_settings ist ON (i.issue_id = ist.issue_id)
               LEFT JOIN issue_settings iss ON (i.issue_id = iss.issue_id AND iss.setting_name = 'crossref::registeredDoi')
            WHERE
                i.published = 1 AND ist.setting_name = 'pub-id::doi' AND ist.setting_value IS NOT NULL
                AND i.journal_id = 1
                AND iss.setting_value IS NULL OR iss.setting_value = ''
                ORDER BY i.date_published DESC 
```

If you change it below, you will also get the unpublished issues...